staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go

```diff
@@ -40,11 +40,10 @@ var PanicHandlers = []func(interface{}){logPanic}
 // called in case of panic.  HandleCrash actually crashes, after calling the
 // handlers and logging the panic message.
 //
-// TODO: remove this function. We are switching to a world where it's safe for
-// apiserver to panic, since it will be restarted by kubelet. At the beginning
-// of the Kubernetes project, nothing was going to restart apiserver and so
-// catching panics was important. But it's actually much simpler for monitoring
-// software if we just exit when an unexpected panic happens.
+// It is recommended to defer HandleCrash in any functions with panic risk.
```

Comment:
Is it recommended? Why not just let it crash? Virtually every function can panic due to nil ptr derefs.